plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace edsl;
+
+namespace {
+
+// Compute pad_before and the size of output.
+// it is based on (plaidml/op/lib/ops.cc.) and i change the type of the variables.
+std::pair<size_t, size_t> compute_padding_before_and_output_size(size_t input_size, size_t off_size, size_t filter_size,
+                                                                 size_t stride, plaidml::op::AutoPadMode autopad_mode,
+                                                                 size_t pad_before, size_t pad_end, size_t dilation) {
+  size_t I_eff = input_size;
+  size_t F_eff = (dilation * (filter_size - 1)) + 1;
+  if (autopad_mode == plaidml::op::AutoPadMode::EXPLICIT) {
+    size_t output_size = ((I_eff + pad_before + pad_end - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::VALID) {
+    size_t output_size = ((I_eff - F_eff + stride) / stride);
+    std::pair<size_t, size_t> result(0, output_size);
+    return result;
+  }
+  if (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER || autopad_mode == plaidml::op::AutoPadMode::SAME_UPPER) {
+    size_t lower_term = (autopad_mode == plaidml::op::AutoPadMode::SAME_LOWER) ? 1 : 0;
+    size_t max = 0;
+    if (((off_size - 1) * stride + F_eff - input_size) > 0) {
+      max = (off_size - 1) * stride + F_eff - input_size;
+    }
+    pad_before = (max + lower_term) / 2;
+    size_t output_size = ((I_eff + stride - 1) / stride);
+    std::pair<size_t, size_t> result(pad_before, output_size);
+    return result;
+  }
+  THROW_IE_EXCEPTION << "Unexpected autopadding mode.";
+}
+
+// Cast long int to int in vector.
+std::vector<int> cast_vector(std::vector<int64_t> vec) {
+  std::vector<int> cast_vec(vec.size());
+  for (auto i = 0; i < vec.size(); ++i) {
+    cast_vec[i] = static_cast<int>(vec[i]);
+  }
+  return cast_vec;
+}
+
+// Extract values which is needed.
```

Comment:
Maybe add the 2d specific example what values are needed, instead of this general description.